src/plugins/intel_gpu/include/intel_gpu/primitives/rnn.hpp

```diff
@@ -0,0 +1,189 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+#include "primitive.hpp"
+#include "activation.hpp"
+#include <vector>
+#include <algorithm>
+#include <string>
+#include "intel_gpu/graph/serialization/activation_serializer.hpp"
+
+namespace cldnn {
+
+/// @brief Weights orders
+/// @details Specifies the order in which the weights are concatenated.
+/// e.g. [i, o, f, z] : [input, output, forget, block]
+/// ONNX order: iofz
+/// Caffe order: ifoz
+/// pyTorch order: izof
+/// OV order: fizo
+enum class lstm_weights_order {
+    iofz,
+    ifoz,
+    izof,
+    fizo
+};
+
+template <typename PType>
+struct RNNParams : public primitive_base<PType> {
+    RNNParams() : primitive_base<PType>("", {}) {}
+    RNNParams(const RNNParams&) = default;
+    RNNParams(const primitive_id& id,
+              const input_info& x,
+              const input_info& initial_hidden_state,
+              const input_info& initial_cell_state,
+              const input_info& W,
+              const input_info& R,
+              const input_info& B,
+              const input_info& seq_lenghts,
+              const float clip = 0,
+              bool input_forget = false,
+              const std::vector<activation_func>& activations = {activation_func::logistic,
+                                                                activation_func::hyperbolic_tan,
+                                                                activation_func::hyperbolic_tan},
+              const std::vector<activation_additional_params>& activation_params = {},
+              const lstm_weights_order& offset_order = lstm_weights_order::iofz,
+              const ov::op::RecurrentSequenceDirection direction = ov::op::RecurrentSequenceDirection::FORWARD,
+              const padding& output_padding = padding(),
```

Comment:
I think padding arg is not needed as it's always set as default

Reply:
done